### ISSUE TITLE
3.7.2 tries to use systemd journal on Ubuntu Trusty by default

### ISSUE THREAD

**Issue by mdebb** [balabit/syslog-ng#987]
Tried to install syslog-ng 3.7.2 on a fresh and clean Ubuntu Trusty machine, but after installation, only syslog-ng internal messages are logged to files, neither kernel, nor user-space logs are present.

The configuration (taken from OSE Admin Guide):
```
@version: 3.7

@include "scl.conf"

source s_local { system(); internal(); };

destination d_local {

            file("/var/log/messages"); };

log { source(s_local); destination(d_local); };
```

The output of --preprocess-into contains the following part:

```
source s_local { channel {
    source {
systemd-journal();

    }; # source
channel {
  channel {
    parser {
      json-parser(prefix('.cim.') marker('@cim:'));
    }};
    flags(final);
  };
  channel { };
};
}; # channel
; internal(); };

```

Part of strace output:
```
uname({sys="Linux", node="trusty", ...}) = 0
gettid()                                = 1441
mmap(NULL, 548864, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f8400795000
openat(AT_FDCWD, "/run/log/journal", O_RDONLY|O_NONBLOCK|O_DIRECTORY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/var/log/journal", O_RDONLY|O_NONBLOCK|O_DIRECTORY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
inotify_init1(IN_NONBLOCK|IN_CLOEXEC)   = 11
openat(AT_FDCWD, "/run/log/journal", O_RDONLY|O_NONBLOCK|O_DIRECTORY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/var/log/journal", O_RDONLY|O_NONBLOCK|O_DIRECTORY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
fcntl(11, F_GETFD)                      = 0x1 (flags FD_CLOEXEC)
fcntl(11, F_GETFL)                      = 0x800 (flags O_RDONLY|O_NONBLOCK)
setsockopt(11, SOL_SOCKET, SO_OOBINLINE, [1], 4) = -1 ENOTSOCK (Socket operation on non-socket)
rename("/var/lib/syslog-ng/syslog-ng.persist-", "/var/lib/syslog-ng/syslog-ng.persist") = 0
open("/var/lib/syslog-ng/syslog-ng.pid", O_WRONLY|O_CREAT|O_TRUNC, 0666) = 12
fstat(12, {st_mode=S_IFREG|0644, st_size=0, ...}) = 0
mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f8400828000
write(12, "1441\n", 5)                  = 5
close(12)                               = 0

```

/etc/lsb-release:
```
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=14.04
DISTRIB_CODENAME=trusty
DISTRIB_DESCRIPTION="Ubuntu 14.04.4 LTS"

```
syslog-ng 3.7.2-2 from the https://build.opensuse.org/project/repositories/home:laszlo_budai:syslog-ng repositories:
```
syslog-ng 3.7.2
Installer-Version: 3.7.2
Revision: 3.7.2-2
Compile-Date: Oct 26 2015 13:09:23
Error opening plugin module; module='mod-java', error='libjvm.so: cannot open shared object file: No such file or directory'
Available-Modules: system-source,redis,json-plugin,afsql,csvparser,mod-python,confgen,afstomp,basicfuncs,dbparser,cryptofuncs,afsmtp,riemann,afuser,graphite,linux-kmsg-format,afprog,pseudofile,syslogformat,afsocket,geoip-plugin,affile,afamqp,afmongodb,sdjournal
Enable-Debug: off
Enable-GProf: off
Enable-Memtrace: off
Enable-IPv6: on
Enable-Spoof-Source: on
Enable-TCP-Wrapper: on
Enable-Linux-Caps: off
```

```
# dpkg -l system\*
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name                              Version               Architecture          Description
+++-=================================-=====================-=====================-========================================================================
un  system-log-daemon                 <none>                <none>                (no description available)
un  system-services                   <none>                <none>                (no description available)
un  systemd                           <none>                <none>                (no description available)
ii  systemd-services                  204-5ubuntu20.18      amd64                 systemd runtime services
ii  systemd-shim                      6-2bzr1               amd64                 shim for systemd
```

**Comment by mdebb** [comment]
After creating a new (oldschool) source: 
`source s_local { file("/proc/kmsg"); unix-dgram("/dev/log"); internal(); };` 
the following is in the output of a debug start (-Fdvte):

```
[2016-03-16T15:35:13.252489] Module loaded and initialized successfully; module='afsocket'
[2016-03-16T15:35:13.252703] Using /dev/log Unix socket with systemd is not possible. Changing to systemd-syslog source, which supports socket activation.;
...
[2016-03-16T15:35:13.257022] Failed to acquire /run/systemd/journal/syslog socket, disabling systemd-syslog source;

```

**Comment by juhaszviktor** [comment]
I think the PR #327 should be ported to the version 3.7 and above

**Comment by bkil-syslogng** [comment]
If you are in a hurry, we recommend compiling from source and configuring with `--disable-systemd`. We will discuss whether it could make it for 3.7.3, or if it will only be available on 3.8.

**Comment by mdebb** [comment]
Thanks, compiling with --disable-systemd produced a good workaround.

**Comment by bkil-syslogng** [comment]
I'm closing this issue because @mdebb got a workaround and the fix in #988 (the backport that @juhaszviktor has mentioned) has also been merged to 3.7/master. We have agreed that the fix will find its way into 3.7.3.

**Comment by bkil-syslogng** [comment]
`scan-build -enable-checker deadcode.DeadStores` would have caught this issue.
